internal/bucket/lifecycle/rule.go

```diff
@@ -158,7 +160,10 @@ func (r Rule) Validate() error {
 	if err := r.validateNoncurrentTransition(); err != nil {
 		return err
 	}
-	if !r.Expiration.set && !r.Transition.set && !r.NoncurrentVersionExpiration.set && !r.NoncurrentVersionTransition.set {
+	if (!r.Filter.Tag.IsEmpty() || len(r.Filter.And.Tags) != 0) && !r.DelMarkerExpiration.Empty() {
+		return errInvalidRuleDelMarkerExpiration
+	}
+	if !r.Expiration.set && !r.Transition.set && !r.NoncurrentVersionExpiration.set && !r.NoncurrentVersionTransition.set && r.DelMarkerExpiration.Empty() {
```

Comment:
I think we can leave the old change of this line as it is and add a new 'if' with r.DeMarkerExpiration involved

Reply:
I'm not sure what you have in mind. Could you explain with a modified `if`. In this if condition I'm trying to disallow ILM rules which don't have at least one of the above actions in it.